src/rust/engine/dep_inference/build.rs

```diff
@@ -155,23 +157,65 @@ pub trait Visitor {
     .unwrap();
 }
 
+fn gen_impl_hash_file(name: &'static str, source_dir: &Path, impl_dir: &Path, out_dir: &Path) {
```

Comment:
I wonder whether this code should include the global `Cargo.lock`, or some subset of it? For instance, if there's bugs in the treesitter libs, we might upgrade them without changing any of this code, but we'd want dev inference to avoid reading old cached entries?

Without fine-grained dependency detection, this will result in spurious cache busting, but that's probably better than the other way around?

Reply:
I ended up embedding the package version in the hash. Then we can bump if we need. Sound good?